From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: pmos-laurel-sprout <dev@pmos.laurel.invalid>
Date: 2026-08-03
Subject: [PATCH] arm64: dts: qcom: sm6125-xiaomi-laurel-sprout: enable MDSS + S6E8FC0 panel

Porta el enable de MDSS/DSI + panel Samsung S6E8FC0 M1906F9 desde
mainline master (commit 493cb869874c) con el compatible CORREGIDO
samsung,s6e8fc0-m1906f9 (el master upstream usa el typo s6e8fco,
ver docs/DTS-AUDIT.md).

Añade:
- reguladores fijos panel_vdd_1p8 (gpio26) y panel_vci_3p0 (gpio124)
- &mdss, &mdss_dsi0 con panel@0, &mdss_dsi0_out (data-lanes),
  &mdss_dsi0_phy status okay
- &dispcc status okay (adición downstream: el commit upstream
  493cb869874c no lo habilita, pero es necesario para MDSS_GDSC y
  los clocks DISP_CC_*; el árbol sm61x5-mainline 7.0-develop lo hace)
- pinctrl mdss_default / mdss_sleep (gpio90) y enables de panel


Signed-off-by: pmos-laurel-sprout <dev@pmos.laurel.invalid>
---
 arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts | 98 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 98 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
--- laurel-v71.dts	2026-08-02 23:24:34.025585711 -0500
+++ laurel-panel.dts	2026-08-02 23:40:24.496381383 -0500
@@ -94,8 +94,35 @@
 		pinctrl-0 = <&ts_vdd_en>;
 		pinctrl-names = "default";
 	};
 
+	panel_vdd_1p8: regulator-panel-vdd {
+		compatible = "regulator-fixed";
+		regulator-name = "panel_vdd_1p8";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		vin-supply = <&vreg_l9a>;
+
+		enable-active-high;
+		gpio = <&tlmm 26 GPIO_ACTIVE_HIGH>;
+		pinctrl-0 = <&panel_vdd_1p8_en>;
+		pinctrl-names = "default";
+	};
+
+	panel_vci_3p0: regulator-panel-vci {
+		compatible = "regulator-fixed";
+		regulator-name = "panel_vci_3p0";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+
+		enable-active-high;
+		gpio = <&tlmm 124 GPIO_ACTIVE_HIGH>;
+		pinctrl-0 = <&panel_vci_3p0_en>;
+		pinctrl-names = "default";
+	};
+
 	thermal-zones {
 		rf-pa0-thermal {
 			thermal-sensors = <&pm6125_adc_tm 0>;
 
@@ -161,8 +188,51 @@
 		touchscreen-size-y = <1560>;
 	};
 };
 
+&dispcc {
+	status = "okay";
+};
+
+&mdss {
+	status = "okay";
+};
+
+&mdss_dsi0 {
+	vdda-supply = <&vreg_l18a>;
+	status = "okay";
+
+	panel@0 {
+		compatible = "samsung,s6e8fc0-m1906f9";
+		reg = <0>;
+
+		reset-gpios = <&tlmm 90 GPIO_ACTIVE_LOW>;
+
+		pinctrl-0 = <&mdss_default>;
+		pinctrl-1 = <&mdss_sleep>;
+		pinctrl-names = "default", "sleep";
+
+		vdd-supply = <&panel_vdd_1p8>;
+		vci-supply = <&panel_vci_3p0>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&mdss_dsi0_out>;
+			};
+		};
+	};
+};
+
+&mdss_dsi0_out {
+	data-lanes = <0 1 2 3>;
+	remote-endpoint = <&panel_in>;
+};
+
+
+&mdss_dsi0_phy {
+	status = "okay";
+};
+
 &pm6125_adc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&camera_flash_therm &emmc_ufs_therm>;
 
@@ -425,8 +495,15 @@
 
 &tlmm {
 	gpio-reserved-ranges = <22 2>, <28 6>;
 
+	panel_vdd_1p8_en: panel-vdd-default-state {
+		pins = "gpio26";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-up;
+	};
+
 	ts_vdd_en: ts-vdd-default-state {
 		pins = "gpio83";
 		function = "gpio";
 		drive-strength = <2>;
@@ -459,8 +536,29 @@
 		function = "gpio";
 		drive-strength = <2>;
 		bias-pull-down;
 	};
+
+	mdss_default: mdss-default-state {
+		pins = "gpio90";
+		function = "gpio";
+		drive-strength = <8>;
+		bias-disable;
+	};
+
+	mdss_sleep: mdss-sleep-state {
+		pins = "gpio90";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-down;
+	};
+
+	panel_vci_3p0_en: panel-vci-default-state {
+		pins = "gpio124";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-up;
+	};
 };
 
 &ufs_mem_hc {
 	vcc-supply = <&vreg_l24a>;

